### PR TITLE
fix: handle exit code correctly

### LINKED ISSUE #191: exit with non-zero code when any step fails
+ Right now, we always exit the process with `0` even when there are step that contains failures. 

```
journey('test', async ({ page }) => {
  step('error step', async () => {
    throw new Error('boom');
  });
});
```

We already capture and show this error on both of our reporters, but we have to exit with non-zero code when `result.status == 'failed'`

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ fix #191 
+ Handle the exit code from the CLI runs correctly, now we exit with >0 error codes when any of the journey fails which include steps as well. If any error is thrown via assertion or error - CLI will exit with 1 exit code
+ Added tests by creating a CLI mock that runs and waits for the exit code.
+ Cleaned up some unwanted CLI args parsing code which is not relevant for our run API.

### CODE REVIEW

File: __tests__/core/browser-service.test.ts [hunk 1]
```diff
@@ -24,26 +24,25 @@
...
     await browserService.dispose();
   });
 
   it('should create browser pages', async () => {
-    const browser = await chromium.connect({
-      wsEndpoint: 'ws://localhost:9323',
-    });
-    const context = await browser.newContext();
-    const page = await context.newPage();
-    await page.goto('https://elastic.co');
```
Comment (vigneshshanmugam) [inline __tests__/core/browser-service.test.ts:46]: missed this in my last review. we use local dev server to test. fixed it now.

File: __tests__/index.test.ts [hunk 2]
```diff
@@ -56,7 +56,7 @@ describe('Run', () => {
   });
 });
 
-describe('Run - cli args', () => {
+describe.skip('Run - cli args', () => {
```
Comment (vigneshshanmugam) [inline __tests__/index.test.ts:59]: These tests tie up our run API with CLI which is incorrect, I will clean this up or will do a follow up PR to get rid of this and add proper tests for CLI for arguments.

File: src/cli.ts [hunk 3]
```diff
@@ -175,6 +175,16 @@ async function prepareSuites(inputs: string[]) {
...
+
+  /**
+   * Exit with error status if any journey fails
+   */
+  for (const result of Object.values(results)) {
+    if (result.status === 'failed') {
+      process.exit(1);
+    }
+  }
+  return results;
```
Comment (vigneshshanmugam) [inline src/cli.ts:187]: note to self: remove

File: src/index.ts [hunk 4]
```diff
@@ -36,34 +35,22 @@ export async function run(options: RunOptions) {
   sourceMapSupport.install({
     environment: 'node',
   });
-
-  const cliArgs = parseArgs();
```
Comment (vigneshshanmugam) [inline src/index.ts:40]: Not sure how this got introduced, but this is wrong. Our CLI already parses and sends the parsed options to our run API.

File: __tests__/index.test.ts (hunk 2 above)
Comment (vigneshshanmugam) [reply __tests__/index.test.ts]: Fixed now.

**Review by andrewvc — APPROVED** [review]
LGTM. Tested manually as well, works as advertised

**Pull request merged**